input-output/pydicom_series.py

```diff
@@ -617,6 +612,13 @@ def _finish(self):
 
         # Init measures to check (these are in 2D)
         dimensions = ds1.Rows, ds1.Columns
+        if (samples := ds1.SamplesPerPixel) > 1:
+            if (conf := ds1.PlanarConfiguration) == 0:
+                dimensions += (samples,)
+            elif conf == 1:
+                dimensions = (samples,) + dimensions
+            else:
+                raise ValueError(f"Invalid Planar Configuration: '{conf}'")
 
```

Comment:
Shouldn't `dimensions2` below also be updated for correct comparison of the two?

Reply:
Oh definitely, thanks!

I have a question that relates also to the previous comment:
Does pydicom's `Dataset` has a method to get the expected `ds.pixel_array.shape`?
The best I could locate with a quick search is [`reshape_pixel_array`](https://github.com/pydicom/pydicom/blob/c8f98eba4c79ad7b53818ba85bcb1e772bfe6bae/pydicom/pixel_data_handlers/util.py#L1160), which internally computes this shape.

I think it's worthwhile to add a `shape` property to `Dataset`, so that a user can call to `ds.shape` without reading `pixel_array`.
Nibabel for example supports this syntax.

Reply:
> Does pydicom's `Dataset` has a method to get the expected `ds.pixel_array.shape`?

Not that I can think of - I think `reshape_pixel_array` has the best treatment of it, including the nice doc string explaining it.  @scaramallion can maybe comment on this, having created that code and generally being deeper into handling different image types.

> I think it's worthwhile to add a `shape` property to `Dataset`, so that a user can call to `ds.shape` without reading `pixel_array`.

 I agree that separating the shape part out as a Dataset property is useful - certainly for this contrib code, but possibly for any time one wants to group different DICOM files before processing pixel data.

Reply:
Does is make sense to add such a property to an object meant only to represent a DICOM dataset? I also think the name `shape` is a bit vague, since we're only talking about datasets with Pixel Data and it's not immediately obvious what sort of 'shape' a dataset would have (other suggestions `array_shape`, `pixel_shape`, `image_shape`, `pixel_array_shape`).

The second issue is that NiBabel is conceptually about images, where it makes sense to talk about a shape. The example literally says:
> A NiBabel image knows about its shape:
```python
>>> img.shape
(128, 96, 24, 2)
```
A third issue is what happens when we have overlays, waveforms and pixel data in the same dataset? What is the expected answer?

Reply:
It's also not particularly difficult to determine the shape yourself:
```python
shape = (ds.get('NumberOfFrames', 1) or 1, ds.Rows, ds.Columns, ds.SamplesPerPixel)
```

Reply:
> Does is make sense to add such a property to an object meant to represent a DICOM dataset? I also think the name `shape` is a bit vague; it's not immediately obvious what sort of 'shape' a dataset would have (other suggestions `pixel_data_shape`, `pixel_array_shape`).

1. I think it does, just as we have the `pixel_array` property. Both are useful.
Specific use cases of the proposed property include: 
    - Shape comparison between slices (just as here)
    - Missing handlers - `ds.pixel_array` cannot be returned, but we can return the shape of it

    The name should be discussed, though I think a name with two underscores is somewhat clumsy.

> The second issue is that NiBabel is conceptually about images, where it makes sense to talk about a shape. The example literally says:
> 
> > A NiBabel image knows about its shape:
> 
> ```python
> >>> img.shape
> (128, 96, 24, 2)
> ```
> 
> This isn't true for _pydicom_ where we are more concerned about the dataset itself.

2. It is true that in _pydicom_ the emphasis is on the whole dataset. The DICOM standard is way richer than NIfTI.
However, the `img` above is nifti dataset (`nib.Nifti1Image`), and the analogy is:

    | Criterion  | _pydicom_ | _NiBabel_ |
    | ------------- | ------------- | ------------- |
    | file format  | DICOM  | NIfTI (+others) |
    | dataset and type | `ds`, `Dataset`  | `img`, `Nifti1Image` (+others) |
    | image array | `ds.pixel_array` | `img.get_fdata()` or `np.asanyarray(img.dataobj)` |
    | image shape | only `ds.pixel_array.shape` | `img.shape` |
    | image dtype | only `ds.pixel_array.dtype` | `img.get_data_dtype()` |

> A third issue is what happens when we have overlays, waveforms and pixel data in the same dataset? What is the expected answer?

3. The expected answer is the expected `ds.pixel_array.shape`.
Overlays (and waveforms?) can be treated elsewhere.

> A fourth issue, what do we return for enhanced datasets like _Enhanced CT_ which may contain an entire series? Functionally such a dataset with 100 images: (100, 1024, 1024, 1) would appear to be be the same as a dataset with 1 image containing 100 frames.

4. So just as above, we return the shape of the pixel array, which can be (100, 1024, 1024) for enhanced/multiframe dicoms.

> It's also not particularly difficult to determine the shape yourself:
> 
> ```python
> shape = (ds.get('NumberOfFrames', 1) or 1, ds.Rows, ds.Columns, ds.SamplesPerPixel)
> ```

5. I tend to disagree, for a few reasons:
    - Even in your one-liner some in-depth knowledge of the dicom standard is incorporated:
      - Why do you call `ds.get('NumberOfFrames', 1) or 1` instead of just `ds.NumberOfFrames`?
      - Why does the number of frames come first and not last, or after the columns?
      
      (I know the answers to those questions, but a user may not be willing to dive into it.)
    - We do not want 1's in the shape - if `ds.SamplesPerPixel` is 1, it should not be there. 
      The same applies for `ds.NumberOfFrames`.
    - It also depends on `ds.PlanarConfiguration`. If it is 1 (and not 0), the expected shape is:
      `(ds.SamplesPerPixel, ds.get('NumberOfFrames', 1) or 1, ds.Rows, ds.Columns)`

-----

@scaramallion @darcymason let's decide here if we want to open an issue about it in _pydicom_ and continue there :)

Reply:
I'd like to think about these issues a bit more.  There are valid comments from multiple viewpoints here.

I often think of something object-oriented, an intermediate object that can handle related image data elements logically all together.  This is related again to the very long-standing discussion like in https://github.com/pydicom/pydicom/issues/58 and https://github.com/pydicom/pydicom/issues/50.

So perhaps something like `ds.image` could be created, that would be a (meta) object within `ds` that could return information like shape, etc., and there could be parallel concepts for overlay images etc.

In any case, I think the function added here solves it for these purposes, and the thinking can continue in mainline pydicom.